core/modules/mysql/MySqlConfig.cc

```diff
@@ -55,23 +55,20 @@ MySqlConfig::MySqlConfig(std::string const& username,
                          std::string const& hostname,
                          unsigned int const port,
                          std::string const& socket,
-                         std::string const& dbName)
+                         std::string const& dbName,
+                         bool const checkValid)
     : username(username), password(password), hostname(hostname), port(port),
       socket(socket), dbName(dbName) {
 
-    if ((hostname.empty() or port == 0) and socket.empty()) {
-        LOGS(getLogger(), LOG_LVL_FATAL, "MySQL hostname:port and socket both undefined");
-        throw std::invalid_argument("MySQL hostname:port and socket both undefined" );
+    if (checkValid) {
+        checkValidity();
```

Comment:
I'm not sure how useful is this. The ultimate check is when you try to open database connection, so you will get a (reasonable) error message then anyway if your parameters are not good. Catching errors here is extra work which will be repeated later anyways.